readthedocs/search/tests/test_views.py

```diff
@@ -0,0 +1,147 @@
+        self._reindex_elasticsearch(es_index=es_index)
+
+    def test_search_by_project_name(self, client, project):
+        result, _ = self._get_search_result(url=self.url, client=client,
+                                            search_params={'q': project.name})
+
+        assert project.name.encode('utf-8') in result.text().encode('utf-8')
+
+    @pytest.mark.parametrize('data_type', ['content', 'headers', 'title'])
+    @pytest.mark.parametrize('page_num', [0, 1])
+    def test_search_by_file_content(self, client, project, data_type, page_num):
+        query = get_search_query(project_slug=project.slug, page_num=page_num,
+                                 data_type=data_type)
+
+        versions = project.versions.all()
+        # There should be only one version of the project
+        assert len(versions) == 1
+        print(query, page_num, data_type)
+
+        result, _ = self._get_search_result(url=self.url, client=client,
+                                            search_params={'q': query, 'type': 'file'})
+        assert len(result) == 1
+
+    def test_file_search_show_projects(self, client):
+        """Test that search result page shows list of projects while searching for files"""
+
+        # `Github` word is present both in `kuma` and `pipeline` files
+        # so search with this phrase
+        result, page = self._get_search_result(url=self.url, client=client,
+                                               search_params={'q': 'GitHub', 'type': 'file'})
+
+        # There should be 2 search result
+        assert len(result) == 2
+
+        # there should be 2 projects in the left side column
+        content = page.find('.navigable .project-list')
+        assert len(content) == 2
+        text = content.text()
+
+        # kuma and pipeline should be there
+        assert 'kuma' and 'pipeline' in text
+
+    def test_file_search_filter_by_project(self, client):
+        """Test that search result are filtered according to project"""
+
+        # `Github` word is present both in `kuma` and `pipeline` files
+        # so search with this phrase but filter through `kuma` project
+        search_params = {'q': 'GitHub', 'type': 'file', 'project': 'kuma'}
+        result, page = self._get_search_result(url=self.url, client=client,
+                                               search_params=search_params)
+
+        # There should be 1 search result as we have filtered
+        assert len(result) == 1
+        content = page.find('.navigable .project-list')
+
+        # kuma should should be there only
+        assert 'kuma' in result.text()
+        assert 'pipeline' not in result.text()
+
+        # But there should be 2 projects in the left side column
+        # as the query is preset in both projects
+        content = page.find('.navigable .project-list')
+        if len(content) != 2:
+            pytest.xfail("failing because currently project list not show")
```

Comment:
This error could be improved. "not show" what?

What is this an expected failure?

Reply:
So the thing is, though you are filtering by a project, you should able to see all the projects matching with the query in the project list. currently, its showing only one project, but it should show both.